contrib/cmus_all.lua

```diff
@@ -0,0 +1,78 @@
+-----------------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2017, JuanKman94 <juan.carlos.menonita@gmail.com>
+-----------------------------------------------------------
+
+-- {{{ Grab environment
+local tonumber = tonumber
+local io = { popen = io.popen }
+local setmetatable = setmetatable
+local string = { gmatch = string.gmatch, format = string.format }
+local helpers = require("vicious.helpers")
+-- }}}
+
+
+-- Cmus: provides CMUS information
+-- vicious.widgets.cmus
+local cmus_all = {}
+
+
+-- {{{ CMUS widget type
+local function worker(format, warg)
+    local cmus_state  = {
+        ["{duration}"]  = 0,
+        ["{file}"] = "N/A",
+        ["{status}"] = "N/A",
+        ["{title}"]  = "N/A",
+        ["{artist}"]  = "N/A",
+        ["{continue}"]  = "off",
+        ["{shuffle}"]  = "off",
+        ["{repeat}"]  = "off",
+    }
+
+    -- Fallback to CMUS defaults
+    local host = warg and (warg.host or warg[1]) or os.getenv("CMUS_SOCKET")
+
+    if not host then
+      if os.getenv("XDG_RUNTIME_DIR") then
+        host = os.getenv("XDG_RUNTIME_DIR") .. "/cmus-socket"
+      else
+        host = os.getenv("HOME") .. "/.config/cmus/socket"
+      end
+    end
+
+    -- Get data from CMUS server
+    local f = io.popen("cmus-remote --query --server " .. helpers.escape(host))
```

Comment:
I took the wrong function, this should have been `helpers.shellquote`.

Reply:
Sorry I couldn't see this earlier